### PR TITLE
Upload pasted images to server

### PR DESCRIPTION
**Pull request by cdt-eth**
## Problem
With the addition of the [new Image Upload feature](https://twitter.com/nounsprophouse/status/1618317159830278144), we came across scenarios where users were copy/pasting images from screenshots tools or from their desktop. This would add the files as [base64-encoded images](https://www.keycdn.com/support/image-base64-encoding#:~:text=Base64%20encoding%20is%20a%20way,a%20filename%20for%20various%20reasons.) to our DB which would cause the proposals to take forever to load and eventually crash the page. 

## Solution
We now handle _**clipboard-pasted**_ files ontop of the newly-supported _**drag event**_ files. Our onPaste function intercepts the images from the clipboard, uploades them to our server, and then pastes in `` tags with the new Piñata links into the editor.

ezgif com-gif-maker (27)

### CODE REVIEW

File: packages/prop-house-webapp/src/components/ImageUploadModal/index.tsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ const ImageUploadModal: React.FC<{
           buildIpfsPath(r.data.ipfsHash),
           Quill.sources.USER,
         );
+        quill.insertText(quill.getSelection()!.index + 1, '\n\n', Quill.sources.USER);
```
Comment (cdt-eth) [inline packages/prop-house-webapp/src/components/ImageUploadModal/index.tsx:78]: ensures that images are added with a newline space vs inline with text.

File: packages/prop-house-webapp/src/components/ProposalInputs/ProposalInputs.module.css [hunk 2]
```diff
@@ -63,3 +61,15 @@
...
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(255, 255, 255, 0.8);
+  z-index: 100;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
```
Comment (cdt-eth) [inline packages/prop-house-webapp/src/components/ProposalInputs/ProposalInputs.module.css:75]: loading state UI

File: packages/prop-house-webapp/src/components/ProposalInputs/index.tsx [hunk 3]
```diff
@@ -106,11 +157,17 @@ const ProposalInputs: React.FC<{
...
                     placeholder={descriptionData.placeholder}
                     onBlur={() => {
                       setEditorBlurred(true);
                     }}
                   />
+                  {loading && (
+                    <div className={classes.loadingOverlay}>
+                      <LoadingIndicator />
+                    </div>
+                  )}
```
Comment (cdt-eth) [inline packages/prop-house-webapp/src/components/ProposalInputs/index.tsx:170]: full page loading UI while images are being uploaded
Screenshot 2023-02-01 at 4 59 26 PM

File: packages/prop-house-webapp/src/components/ProposalInputs/index.tsx [hunk 4]
```diff
@@ -43,6 +47,53 @@ const ProposalInputs: React.FC<{
     client.current = new PropHouseWrapper(host, signer);
   }, [signer, host]);
 
+  const [loading, setLoading] = useState<boolean>(false);
+  const signerless = new PropHouseWrapper('https://prod.backend.prop.house');
```
[the remark this answers is not in the document]
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/ProposalInputs/index.tsx]: done! https://github.com/Prop-House/prop-house-monorepo/pull/418/commits/a6c8d8de6b2872442840983627f8fd002c823f78

**Review by cryptoseneca — APPROVED** [review]

**Pull request merged**